src/components/astra-editor/index.ts

```diff
@@ -0,0 +1,351 @@
+        }  
+      `
+    }
+
+    const styleElement = document.createElement('style')
+    styleElement.innerHTML = `
+      .cm-tooltip-autocomplete ul::-webkit-scrollbar,
+      .cm-scroller::-webkit-scrollbar {
+        width: 10px;
+        height: 10px;
+      }
+
+      .cm-scroller::-webkit-scrollbar-thumb {
+        border-radius: 5px;
+      }
+
+      .cm-tooltip-autocomplete ul::-webkit-scrollbar-thumb,
+      .cm-scroller::-webkit-scrollbar-thumb {
+        background: rgba(0, 0, 0, 0.3);
+      }
+
+      .dark .cm-tooltip-autocomplete ul::-webkit-scrollbar-thumb, 
+      .dark .cm-scroller::-webkit-scrollbar-thumb {
+        background: rgba(255, 255, 255, 0.3);
+      }
+
+      .cm-tooltip-autocomplete > ul > li {
+        display: flex;
+      }
+
+      .cm-tooltip-autocomplete > ul > li > .cm-completionIcon {
+        width: 1em !important;
+        display: flex;
+        align-self: center;
+        justify-content: center;
+      }
+
+      .cm-tooltip-autocomplete .cm-completionLabel {
+        flex-grow: 1;
+      }
+
+      .cm-tooltip-autocomplete .cm-completionDetail {
+        padding-left: 15px;
+      }
+      
+      .cm-completionIcon-property::after {
+        content: "\\61" !important;
+        font-family: 'outerbase-icon' !important;
+      }
+
+      .cm-completionIcon-type::after {
+        content: "\\62" !important;
+        font-family: 'outerbase-icon' !important;
+      }
+
+      .cm-completionIcon-function::after,
+      .cm-completionIcon-method::after,
+      .cm-completionIcon-variable::after,
+      .cm-completionIcon-namespace::after,
+      .cm-completionIcon-interface::after {
+        content: '⚡' !important;
+      }
+    `
+
+    this.shadowRoot?.append(styleElement)
+
+    this.container = container
+  }
+
+  connectedCallback(): void {
+    this._color = this.getAttribute('color') === 'dark' ? 'dark' : 'light'
+    this._theme = this.getAttribute('theme') ?? 'moondust'
+    this._placeholder = this.getAttribute('placeholder') ?? ''
```

Comment:
we could use `@property()`instead of managing this stuff manually https://lit.dev/docs/components/properties/

which I see you're already doing in other spots. Is there a reason to have both `this.color` and `this._color`?

Reply:
`_color` is the internal state that will be used for `color` getter. Each property has special setter because we need to dispatch different action to code mirror based on type of property.